whisper.cpp

```diff
@@ -5452,10 +5460,11 @@ int whisper_full_with_state(
                         if (decoder.completed || decoder.failed) {
                             continue;
                         }
-
+                        kv_mutex.lock();
                         whisper_kv_cache_seq_rm(state->kv_self, j,                           -1, -1);
                         whisper_kv_cache_seq_cp(state->kv_self, WHISPER_MAX_DECODERS + j, j, -1, -1);
                         whisper_kv_cache_seq_rm(state->kv_self, WHISPER_MAX_DECODERS + j,    -1, -1);
+                        kv_mutex.unlock();
```

Comment:
Where is multithreading used in the code you referred to?

Reply:
You're right, this is not the root cause, it's just because the voice can come out normally after I made the modification, so I didn't look carefully. It looks like I haven't found the root cause. Still investigating.

Reply:
<img width="1399" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/66455008/d0047868-e1fe-4f51-a1cf-734afe3d0df6">